zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/DeploymentReconstructProcessor.java

```diff
@@ -154,7 +208,35 @@ private Set<Resource> findResourcesWithDeploymentKey(
           return form.getTenantId().equals(tenantId) && formDeploymentKey == deploymentKey;
         });
 
-    // TODO: Continue with decisionState
+    decisionState.forEachDecision(
```

Comment:
It actually does work now because you are using a HashSet for the resources 😅 Not sure if this was intended. It's unecessary to do all the extra state lookups.

Reply:
Yeah I did not consider this :facepalm: 
I'll rework this a bit, I don't want to rely on the `Set`.

Reply:
Yeah so iterating through DRGs first made the code easier to read too. The only drawback (at least how I implemented it) is that we have to get all decisions for a DRG even when none of them have a matching decision key. I think that's fine though.